src/lib/containers/entity/annotations/SchemaDrivenAnnotationEditor.tsx

```diff
@@ -0,0 +1,352 @@
+  liveValidate?: boolean
+  onSuccess?: () => void
+}
+
+export type SchemaDrivenAnnotationEditorModalProps = {
+  entityId: string
+  show: boolean
+  onHide: () => void
+}
+
+// patternProperties lets us define how to treat additionalProperties in a JSON schema by property name
+// here we can ban properties that collide with entity properties by making their schema "not: {}"
+const patternPropertiesBannedKeys = entityJsonKeys.reduce((current, item) => {
+  current[`^${item}$`] = { not: {} }
+  return current
+}, {})
+
+function getStandardEntityFields(json: EntityJson): EntityJson {
+  return pick(json, entityJsonKeys) as EntityJson
+}
+
+function removeStandardEntityFields(json: EntityJson): Record<string, unknown> {
+  return omit(json, entityJsonKeys)
+}
+
+/**
+ * Renders a form for editing an entity's annotations. The component also supports supplying just a schema ID,
+ * but work to support annotation flows without an entity (i.e. before creating entities) is not yet complete.
+ */
+export const SchemaDrivenAnnotationEditor: React.FunctionComponent<SchemaDrivenAnnotationEditorProps> = ({
+  entityId,
+  schemaId,
+  liveValidate = false,
+  onSuccess = () => {
+    /* no-op */
+  },
+}: SchemaDrivenAnnotationEditorProps) => {
+  const formRef = useRef<Form<Record<string, unknown>>>(null)
+
+  // If fetching an entity, store the non-annotation fields in this object
+  const [entityJson, setEntityJson] = React.useState<
+    Record<string, unknown> | undefined
+  >(undefined)
+
+  // Annotation fields fetched and modified via the form
+  const [formData, setFormData] = React.useState<
+    Record<string, unknown> | undefined
+  >(undefined)
+
+  // Client-side validation errors
+  const [validationError, setValidationError] = React.useState<
+    AjvError[] | undefined
+  >(undefined)
+
+  // Errors from the backend response
+  const [submissionError, setSubmissionError] = React.useState<
+    SynapseClientError | undefined
+  >(undefined)
+  const [showSubmissionError, setShowSubmissionError] = React.useState(false)
+
+  const [showConfirmation, setShowConfirmation] = React.useState(false)
+
+  const ANNOTATION_EDITOR_TOOLTIP_ID = 'AnnotationEditorTooltip'
+
+  const { data: json, refetch: refetchJson } = useGetJson(entityId!, {
+    enabled: !entityId || !formData, // once we have data, don't refetch. it would overwrite the user's entries
+  })
+
+  useEffect(() => {
+    if (json) {
+      /**
+       * To only show annotations in the form, we must remove non-annotation fields.
+       * We will need to submit these values when we create the entity, so we set them aside
+       * and will merge objects later.
+       */
+
+      setEntityJson(getStandardEntityFields(json))
+      setFormData(removeStandardEntityFields(json))
+    }
+  }, [json])
+
+  const { data: schema, isLoading: isLoadingBinding } = useGetSchemaBinding(
+    entityId!,
+    {
+      enabled: !!entityId,
+      retry: false,
+      refetchOnWindowFocus: false,
```

Comment:
We don't want to retry/refetchOnFocus. Otherwise we will hit a loading state on each refetch if there is no schema.